Move datasets into model registry W&B

sarzhann opened the pull request:
* All files from `data/` are moved into https://wandb.ai/clarify/climate-futures-clarify/artifacts/dataset/sea_temperature/v0
* Implemented model registry manager, which can pull/push datasets from W&B
* The only requirement is to set the `WANDB_TOKEN` env variable before execution.

Commit:
move datasets into model registry W&B

sarzhann commented:
@olewu you can consider this change as a helper utility for moving datasets outside of the package into W&B.
So, it can be used inside your work, when the training step is automated and can be done on our side.

olewu commented on src/forecast_clarify/model_registry.py:
```diff
@@ -0,0 +1,45 @@
+import wandb
```
do we need to include this in the requirements?

sarzhann replied:
great catch, I'll add it

sarzhann replied:
fixed.

olewu commented on requirements.txt:
```diff
@@ -9,3 +9,4 @@ numpy==1.21.0 #1.23.0
 pandas==1.3.5 #1.4.3
 scikit-learn==1.0.2 #1.1.2
 xarray==0.20.2 #2022.3.0
```
I noticed that when I install the requirements only, I cannot load the datasets (e.g. in the example notebook). The issue seems to be that xarray is missing a backend for the netcdf files. I can fix this with `pip install netCDF4`. Do you run into the same issue? If so, maybe we have to add netCDF4 to the requirements.

sarzhann replied:
I didn't see that issue, but probably it's fine to have `netCDF4` in the requirements.txt as well

olewu commented on tests/test_config.py:
```diff
@@ -1,9 +1,17 @@
-from forecast_clarify.config import dirs
+from forecast_clarify.src.forecast_clarify.model_registry import Manager
```
does this work with just `from forecast_clarify.model_registry import Manager`?

olewu committed:
add netCDF4 as requirement

Signed-off-by: Ole <owul@norceresearch.no>

olewu committed:
removed duplicate requirement netCDF4

Didn't see that this was already part of the requirements. This shouldn't be the issue then, so I reverted the change

Signed-off-by: Ole <owul@norceresearch.no>
